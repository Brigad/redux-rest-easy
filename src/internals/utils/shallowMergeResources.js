import isPlainObject from 'lodash.isplainobject';
import mergeWith from 'lodash.mergewith';

const shallowMergeResources = (state, normalizedPayload) =>
  mergeWith(
    state.resources,
    normalizedPayload,
    (resources, newResources) =>
      isPlainObject(resources) ? { ...resources, ...newResources } : undefined,
  );

export default shallowMergeResources;
