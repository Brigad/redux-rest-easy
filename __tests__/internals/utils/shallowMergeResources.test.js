import shallowMergeResources from '../../../src/internals/utils/shallowMergeResources';

const PREVIOUS_STATE = {
  resources: {
    eggs: {
      1: {
        a: 1,
      },
    },
    fruits: {
      1: {
        a: 1,
        b: 2,
      },
      2: {
        a: 1,
        b: 2,
      },
    },
    vegetables: {
      1: {
        a: 1,
        b: 2,
        c: 3,
      },
      2: {
        a: 1,
        b: 2,
        c: 3,
      },
      3: {
        a: 1,
        b: 2,
        c: 3,
      },
    },
  },
};

const NORMALIZED_PAYLOAD = {
  eggs: {
    1: {
      a: 2,
    },
  },
  fruits: {
    1: {
      a: 1,
    },
    2: {
      c: 3,
    },
  },
  vegetables: {
    4: {
      a: 1,
    },
  },
};

describe('shallowMergeResources', () => {
  test('only path', () => {
    expect(
      shallowMergeResources(PREVIOUS_STATE, NORMALIZED_PAYLOAD),
    ).toMatchSnapshot();
  });
});
