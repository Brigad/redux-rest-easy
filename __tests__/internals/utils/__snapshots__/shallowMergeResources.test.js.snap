// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallowMergeResources only path 1`] = `
Object {
  "eggs": Object {
    "1": Object {
      "a": 2,
    },
  },
  "fruits": Object {
    "1": Object {
      "a": 1,
    },
    "2": Object {
      "c": 3,
    },
  },
  "vegetables": Object {
    "1": Object {
      "a": 1,
      "b": 2,
      "c": 3,
    },
    "2": Object {
      "a": 1,
      "b": 2,
      "c": 3,
    },
    "3": Object {
      "a": 1,
      "b": 2,
      "c": 3,
    },
    "4": Object {
      "a": 1,
    },
  },
}
`;
